pkg/component/controller/kubeletconfig.go

```diff
@@ -129,6 +134,17 @@ func (k *KubeletConfig) createProfiles(clusterSpec *v1beta1.ClusterConfig) (*byt
 	defaultProfile := getDefaultProfile(dnsAddress, clusterSpec.Spec.Network.DualStack.Enabled, clusterSpec.Spec.Network.ClusterDomain)
 	defaultProfile["cgroupsPerQOS"] = true
 	defaultProfile["resolvConf"] = "{{.ResolvConf}}"
+	if len(k.Taints) > 0 {
+		var taints []corev1.Taint
+		for _, taint := range k.Taints {
+			parsedTaint, err := parseTaint(taint)
+			if err != nil {
+				return nil, fmt.Errorf("can't parse taints for profile config map: %v", err)
+			}
+			taints = append(taints, parsedTaint)
+		}
+		defaultProfile["registerWithTaints"] = taints
```

Comment:
I'm not sure I get how this would work 🤔 

So we start a controller with `--enable-worker` so we get taints. Now these taints would get populated to `KubeletConfig`. Now if I start a plain worker with a default profile, it'll actually get the same taints from the config, right?

I don't think we can do any of this on the controller side, we need to handle all the kubelet config "mangling" on worker side AFTER we've loaded the "template" from API. I mean a controller setting up the kubelet config `ConfigMap` cannot know which taints etc one wants to put on a worker.

Reply:
Ok, I see. I mixed up the things because the controller also writes kubelet config on disk. 
Will move it to the right place

Reply:
> controller also writes kubelet config on disk
I _think_ it only writes it in case it also starts the worker